ci/build.sh

```diff
@@ -1,166 +1,24 @@
-  popd || exit 1
-}
-
-function bundle_fluentd_plugins() {
-  local version="${1}"
-
-  if [[ -z "${version}" ]] ; then
-    echo "Please provide the version when bundling fluentd plugins"
-    exit 1
-  fi
-
-  find . -maxdepth 1 -name 'fluent-plugin-*' -type 'd' -print |
-    while read -r line; do
-      # Run tests in their own context
-      (bundle_fluentd_plugin "$(basename "${line}")" "${version}") || exit 1
-    done
-}
-
-function set_up_github() {
-  local token="${1}"
-  local branch="${2}"
-
-  git config --global user.name "Continuous Integration"
-  git remote add origin-repo "https://${token}@github.com/SumoLogic/sumologic-kubernetes-collection.git" > /dev/null 2>&1
-  git fetch --unshallow origin-repo
-
-  readonly branch="$(get_branch_to_checkout)"
-  echo "Checking out the ${branch} branch..."
-  git checkout "${branch}"
-}
-
-function push_docker_image() {
-  local version="$1"
-
-  echo "Tagging docker image ${DOCKER_TAG}:local with ${DOCKER_TAG}:${version}..."
-  docker tag "${DOCKER_TAG}:local" "${DOCKER_TAG}:${version}"
-  echo "Pushing docker image ${DOCKER_TAG}:${version}..."
-  echo "${DOCKER_PASSWORD}" | docker login -u "${DOCKER_USERNAME}" --password-stdin
-  docker push "${DOCKER_TAG}:${version}"
-}
 
-function push_helm_chart() {
-  local version="$1"
-  local chart_dir="$2"
-  local sync_dir="./tmp-helm-sync"
+# shellcheck disable=SC1090
+source "$(dirname "${BASH_SOURCE[0]}")/_build_functions.sh"
 
-  echo "Pushing new Helm Chart release ${version}"
-  set -x
-
-  git checkout -- .
-
-  # due to helm repo index issue: https://github.com/helm/helm/issues/7363
-  # we need to create new package in a different dir, merge the index and move the package back
-  mkdir -p "${sync_dir}"
-  helm package deploy/helm/sumologic --dependency-update --version="${version}" --app-version="${version}" --destination "${sync_dir}"
-
-  git fetch origin-repo
-  git checkout gh-pages
-
-  helm repo index --url "https://sumologic.github.io/sumologic-kubernetes-collection${chart_dir:1}/" --merge "${chart_dir}/index.yaml" "${sync_dir}"
-
-  mv -f "${sync_dir}"/* "${chart_dir}"
-  rmdir "${sync_dir}"
-
-  git add -A
-  git commit -m "Push new Helm Chart release ${version}"
-  git push --quiet origin-repo gh-pages
-  set +x
-}
-
-function build_docker_image() {
-  local tag
-  local no_cache
-  tag="${1}"
-
-  echo "Building docker image with ${tag}:local in $(pwd)..."
-  pushd ./deploy/docker || exit 1
-  cp ../helm/sumologic/conf/setup/main.tf . || exit 1
-  no_cache="--no-cache"
-  if [[ "${DOCKER_USE_CACHE}" == "true" ]]; then
-    no_cache=""
-  fi
-  docker build . -f ./Dockerfile -t "${tag}:local" ${no_cache:+"--no-cache"}
-  rm main.tf
-  rm -f ./gems/*.gem
-  popd || exit 1
+if is_checkout_on_tag; then
+  VERSION="$(git describe --tags --always)"
+  readonly VERSION="${VERSION#v}"
+else
+  # Don't use non-tag 'git describe' output as this will
+  # fail when building fluentd plugins.
+  readonly VERSION="0.0.0"
```

Comment:
Didn't we tested that the plugins are build correctly?

Reply:
Argh, I've removed the old build from GH actions but `git describe ...` returned short SHA which `gem bundle ...` complained about.

Reply:
so git describe doesn;t return `1.3.2-1234567`?

Reply:
Ah, ok so it didn't but I've added the `git fetch --tags --prune --unshallow` and then I've figured out that we don't really need to fetch full history (via `git fetch --unshallow ...`) but it's enough to fetch current branch (and make sure we have `unshallow`ed repo history

This is handled in `fetch_current_branch` in here: 

https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/5735894346043a74c008733dcb74da34566e6b44/ci/_build_functions.sh#L126-L143

Reply:
And besides bandwidth we also save time as `git fetch --tags --unshallow` takes roughly 10-20s https://github.com/SumoLogic/sumologic-kubernetes-collection/runs/1464248561?check_suite_focus=true#step:5:1
while fetching only specified branch it takes about a second https://github.com/SumoLogic/sumologic-kubernetes-collection/runs/1470602457?check_suite_focus=true#step:6:5 (till line 72)

Reply:
So we can simplify the if else statement now